src/tools/tidy/src/deps.rs

```diff
@@ -276,6 +277,269 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "zerovec-derive",
 ];
 
+const PERMITTED_CARGO_DEPENDENCIES: &[&str] = &[
```

Comment:
IIUC, this is somewhat doing what rustc_codegen_cranelift does: checking licenses and permitted dependencies for. If cargo is going to check that inside its own repository, we can completely remove this and `EXCEPTIONS_CARGO`.

However, IO feel like we still need `check_rustfix` to exist, as it check the version of rustfix in `compiletest` and `cargo`.

Reply:
The licenses definitely need to be checked here, but I don't think the `PERMITTED` list needs to be included. That was just a technique used by the compiler team to avoid accidentally adding new dependencies that they weren't expecting to add. I don't think cargo needs to bother with that. All the other stuff seems good to keep.

Reply:
Make sense to me! The dependency check has been removed with commit aff1b94. I also found `check_crate_duplicate` seems useless so remove it as well with 0be5193.